crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -1351,10 +1354,34 @@ def upgrade_compute_nodes(virt)
         # continue with that one.
         compute_nodes.sort! { |n| n.upgrading? ? -1 : 1 }
 
-        # This part must be done sequentially, only one compute node can be upgraded at a time
-        compute_nodes.each do |n|
-          upgrade_compute_node(controller, n)
+        while compute_nodes.any?
+          # 1. Evacuate as many as possible, create a subset of nodes without instances.
+          nodes_to_upgrade = []
+          compute_nodes.each do |node|
+            hostname = node[:hostname]
+            begin
+              save_nodes_state(nodes_to_upgrade + [node], "compute", "upgrading")
+              live_evacuate_compute_node(controller, hostname)
+              nodes_to_upgrade << node
+            rescue StandardError
```

Comment:
Well, this is problematic. If `live_evacuate_compute_node` fails for some other reason than we expect (= no free host) user may never find out.

Reply:
I've tried to address this in later commits